jax/numpy/lax_numpy.py

```diff
@@ -2617,6 +2617,34 @@ def polyadd(a, b):
 
   return lax.add(a,b)
 
+def _trim_zeros(a):
+  first = 0
+  for i in a:
+    if i != 0:
+      break
+    first = first + 1
+  return a[first:]
+
+_LEADING_ZEROS_DOC="""\
+Setting trim_leading_zeros=True makes the output match that of numpy.
+But prevents the function from being able to be used in compiled code.
+"""
+
+@_wraps(np.polymul, lax_description=_LEADING_ZEROS_DOC)
+def polymul(a1, a2, *, trim_leading_zeros=False):
+  if isinstance(a1, np.poly1d):
+    a1 = asarray(a1)
+  if isinstance(a2, np.poly1d):
+    a2 = asarray(a2)
+  if len(a1) == 0:
+    a1 = asarray([0.])
+  if len(a2) == 0:
+    a2 = asarray([0.])
```

Comment:
Move this to after the trim_zeros step, or it will be undone :grinning:

Reply:
I might be wrong here. But since it is a length-1 array, it will not completely pass the ```trim_leading_zeros``` check, and hence would be fine?

Reply:
But yeah if I make it after trim_zeros_step, I can probably make the trim_zeros_condition less cluttered.

Reply:
I think if I move it after trim_zeros step and remove the check which says it requires arrays of length greater than one, then ```def _trim_zeros``` will error out.

And if I keep the condition, then there will be no difference.

Reply:
If we don't put the empty array check at the end, think about what happens if the input is `a=[0, 0]` and `trim_leading_zeros=True`. The length is greater than 1, so all zeros will be trimmed, leaving an empty array, which will cause an error in the convolution.

Reply:
Oh I am sorry, I see your point